### PR TITLE
add opa generator powered by OpenAI

### PR DESCRIPTION
**Pull request by vitorvezani**
This PR fixes internal FWI-4322

## Checklist
* [X] I have signed the CLA
* [X] I have updated/added any relevant documentation

### What's the goal of this PR?
- Add OPA policy generator powered by OpenAI
  - Add pre-notice
  - Add in-notice
- Documentation PR is [here](https://github.com/FairwindsOps/insights-docs/pull/330)
  -  Add docs and notice

### CODE REVIEW

File: pkg/cli/generate_opa_openai.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+const (
+	prePolicyNotice = "NOTICE: The OpenAI integration is available for your convenience. Please be aware that you are using your OpenAI API key and all interaction will be governed by your agreement with OpenAI."
+	inPolicyNotice  = "# NOTICE: This policy was generated in part with OpenAI’s large-scale language-generation model. The generated policy should be reviewed and tested for accuracy and revised in order to obtain the desired outcome. The User is responsible for the accuracy of the policy."
+)
+
+var openAIAPIKey, openAIModel, openAIPrompt string
+
+// insights-cli generate opa open-ai
+func init() {
+	generateOPAOpenAI.Flags().StringVarP(&openAIAPIKey, "openai-api-key", "k", "", "The API key for OpenAI")
```
Comment (rbren) [inline pkg/cli/generate_opa_openai.go:23]:
```suggestion
	generateOPAOpenAI.Flags().StringVarP(&openAIAPIKey, "openai-api-key", "k", "", "The API key for OpenAI. Can also be set with environment variable OPENAI_API_KEY")
```

File: pkg/cli/generate_opa_openai.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+	generateOPAOpenAI.Flags().StringVarP(&openAIModel, "openai-model", "m", "gpt-4", "The OpenAI model to use")
+	generateOPAOpenAI.Flags().StringVarP(&openAIPrompt, "prompt", "p", "", "A text specification of the desired OPA policy")
+	generateOPACmd.AddCommand(generateOPAOpenAI)
+}
+
+// generateOPAOpenAI represents the validate opa command
+var generateOPAOpenAI = &cobra.Command{
+	Use:   "openai [-m model] [-k key] -p prompt",
+	Short: "generate an Insights OPA policy based on OpenAI.",
+	Long:  `generate an Insights OPA policy based on OpenAI.`,
```
Comment (rbren) [inline pkg/cli/generate_opa_openai.go:33]:
```suggestion
	Short: "generate an Insights OPA policy with the help of OpenAI's ChatGPT.",
	Long:  `use OpenAI's ChatGPT API to generate an OPA policy by providing an English-language prompt describing what the policy should do. You will need to provide your own OpenAI API key.`,
```

File: pkg/openai/openai.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+}
+
+func (cs *Client) SendPrompt(ctx context.Context, prompt string) (string, error) {
+	message := Message{Role: "user", Content: promptBoilerplate + prompt + "'"}
+	request := Request{Model: cs.model, Messages: []Message{message}}
+	content, err := sendRequest(ctx, cs.apiKey, request)
+	if err != nil {
+		return "", err
+	}
+	return content, nil
```
Comment (rbren) [inline pkg/openai/openai.go:53]: Are we not cleaning up the response at all? Usually GPT replies with some english about what's going on, but I haven't seen that in our output.

**Review by rbren — APPROVED** [review]

File: pkg/openai/openai.go (hunk 3 above)
Comment (vitorvezani) [reply pkg/openai/openai.go:53]: Hmm... actually, it was not necessary - at least by the examples that I've tried, do you want to give it a try to see if the AI will respond with more than the policy?

File: pkg/openai/openai.go [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        "title": "Bad label is present",
+        "description": sprintf("Label %v is present", [found]),
+        "severity": 0.1,
+        "remediation": "Remove the label",
+        "category": "Reliability"
+    }
+}
+<>
+
+The new policy you write should: '`
```
Comment (rbren) [inline pkg/openai/openai.go:36]: Is there an extra singlequote here?
```suggestion
The new policy you write should: `
```

**Pull request merged**